homeassistant/config_entries.py

```diff
@@ -271,43 +282,43 @@ def __init__(
     ) -> None:
         """Initialize a config entry."""
         # Unique id of the config entry
-        self.entry_id = entry_id or uuid_util.random_uuid_hex()
+        object.__setattr__(self, "entry_id", entry_id or uuid_util.random_uuid_hex())
 
         # Version of the configuration.
         self.version = version
```

Comment:
I think version and minor version should also be included in `async_update_entry` so there are no exceptions from the rule about how to modify a config entry. Not sure if there are other attributes that need to be modified by integrations.

Reply:
Implemented in https://github.com/home-assistant/core/pull/110077